playbooks/files/rax-maas/plugins/bonding_iface_check.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+# Copyright 2018, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+
+from maas_common import metric_bool
+from maas_common import print_output
+
+import netiface_check_lib
+
+
+def bonding_ifaces_check():
+    bonding_ifaces = os.lisdir("/proc/net/bonding")
+    for bonding_iface in bonding_ifaces:
```

Comment:
I've noticed that some environments might have a bond file located there, however, it's empty and contains no sub-interfaces. Will this fail to iterate over causing an alarm or just negate the output altogether?

Reply:
This would just produce no metric (skip the iteration) and generate no alarns